File: language/src/main/java/brj/nodes/CollNode.java
package brj.nodes;

import brj.BridjeLanguage;
import com.oracle.truffle.api.dsl.NodeChild;
import com.oracle.truffle.api.dsl.NodeField;
import com.oracle.truffle.api.source.SourceSection;

@NodeChild(value = "els", type = ExecuteArrayNode.class)
@NodeField(name = "sourceSection", type = SourceSection.class)
public abstract class CollNode extends ExprNode {
    protected CollNode(BridjeLanguage lang) {
        super(lang);
    }

    abstract Object executeColl(Object[] res);
}

File: language/src/main/kotlin/brj/nodes/CollNode.kt
package brj.nodes

import brj.BridjeLanguage
import com.oracle.truffle.api.dsl.NodeChild
import com.oracle.truffle.api.source.SourceSection

@NodeChild(value = "els", type = ExecuteArrayNode::class)
abstract class CollNode(lang: BridjeLanguage, loc: SourceSection?) : ExprNode(lang, loc) {
    abstract fun executeColl(els: Array<Any>): Any
}